Content.Server/Atmos/EntitySystems/GasAnalyzerSystem.cs

```diff
@@ -1,46 +1,276 @@
-﻿using Content.Server.Atmos.Components;
+﻿using System.Linq;
+using Content.Server.Atmos;
+using Content.Server.Atmos.Components;
+using Content.Server.NodeContainer;
+using Content.Server.NodeContainer.Nodes;
 using Content.Server.Popups;
+using Content.Shared.Atmos;
+using Content.Shared.Atmos.Components;
 using Content.Shared.Interaction;
+using Content.Shared.Interaction.Events;
 using JetBrains.Annotations;
 using Robust.Server.GameObjects;
 using Robust.Shared.Player;
+using Robust.Shared.Utility;
+using static Content.Shared.Atmos.Components.SharedGasAnalyzerComponent;
 
 namespace Content.Server.Atmos.EntitySystems
 {
     [UsedImplicitly]
     public sealed class GasAnalyzerSystem : EntitySystem
     {
         [Dependency] private readonly PopupSystem _popup = default!;
+        [Dependency] private readonly AtmosphereSystem _atmo = default!;
+        [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+        [Dependency] private readonly UserInterfaceSystem _userInterface = default!;
+
+        private HashSet<EntityUid> _activeAnalyzers = new();
+        private RemQueue<EntityUid> _invalidAnalyzers = new();
 
         public override void Initialize()
         {
             base.Initialize();
 
             SubscribeLocalEvent<GasAnalyzerComponent, AfterInteractEvent>(OnAfterInteract);
+            SubscribeLocalEvent<GasAnalyzerComponent, GasAnalyzerRefreshMessage>(OnRefreshMessage);
+            SubscribeLocalEvent<GasAnalyzerComponent, GasAnalyzerDisableMessage>(OnDisabledMessage);
+            SubscribeLocalEvent<GasAnalyzerComponent, ComponentShutdown>(OnComponentShutdown);
+            SubscribeLocalEvent<GasAnalyzerComponent, DroppedEvent>(OnDropped);
+            SubscribeLocalEvent<GasAnalyzerComponent, UseInHandEvent>(OnUseInHand);
         }
 
         public override void Update(float frameTime)
         {
-            foreach (var analyzer in EntityManager.EntityQuery<GasAnalyzerComponent>(true))
+            if (!_activeAnalyzers.Any())
+                return;
```

Comment:
this isnt necessary since the foreach will just skip over if it has none (and I believe this is incorrect since it should still be processing invalid analyzers?)